service/collector/helper.go

```diff
@@ -0,0 +1,71 @@
+package collector
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type helperConfig struct {
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+type helper struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+}
+
+func newHelper(config helperConfig) (*helper, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	h := &helper{
+		g8sClient: config.G8sClient,
+		logger:    config.Logger,
+	}
+
+	return h, nil
+}
+
+func (h *helper) getCharts() ([]*chartState, error) {
```

Comment:
I think this is not useful code. Why can't we simply list and use the results? As a general rule, as soon as we need such a helper thing, it smells. So we should think twice and have good reasons. The other version I saw where we tried to generalize the provider specific CR information was quite legit because the design this helper version tried to fix was already broken. Here we add a new layer which does not solve any problem.

Reply:
That's a good point, @rossf7 Can we remove this helper function totally and go on with using general APIClient:?

Reply:
Let's just go for it. <3